rpc/client.go

```diff
@@ -31,6 +33,8 @@ type RpcClient struct {
 	completedObjectives   *safesync.Map[chan struct{}]
 	ledgerChannelUpdates  *safesync.Map[chan query.LedgerChannelInfo]
 	paymentChannelUpdates *safesync.Map[chan query.PaymentChannelInfo]
+	cancel                context.CancelFunc
+	wg                    sync.WaitGroup
```

Comment:
Instead of using multiple cancel functions we could pass the same cancealable context into multiple goroutines. This way the single `cancel` function could stop multiple goroutines, which I think would be a bit neater.

Reply:
Yes that makes much more sense. In that model, the context and the cancel and the wait group are all global things.